### PR TITLE
Create Script to Generate CADL

### PR DESCRIPTION
**Pull request by alzimmermsft**
Adds a PowerShell script, `Generate-Cadl.ps1`, to the root of the repository which handles all tasks required to generate CADL except the Maven installation requirements.

### CODE REVIEW

File: Generate-Cadl.ps1 [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Use case:
```
Comment (haolingdong-msft) [inline Generate-Cadl.ps1:1]: Better rename the file name as `Generate-TypeSpec.ps1`

File: Generate-Cadl.ps1 [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+}
+
+Write-Host "Changing directory to './cadl-extension'"
+Set-Location ./cadl-extension
+
+Write-Host "Installing dependencies for CADL Java ('npm install')"
+invokeExpressionAndCaptureOutput("npm install")
+
+Write-Host "Building CADL Java ('npm run build')"
```
Comment (haolingdong-msft) [inline Generate-Cadl.ps1:23]:
```suggestion
Write-Host "Building TypeSpec Java ('npm run build')"
```

File: Generate-Cadl.ps1 [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+invokeExpressionAndCaptureOutput("npm run build")
+
+Write-Host "Linting CADL Java ('npm run lint')"
```
Comment (haolingdong-msft) [inline Generate-Cadl.ps1:26]:
```suggestion
Write-Host "Linting TypeSpec Java ('npm run lint')"
```

File: Generate-Cadl.ps1 [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+invokeExpressionAndCaptureOutput("npm run lint")
+
+Write-Host "Checking CADL Java format ('npm run check-format')"
```
Comment (haolingdong-msft) [inline Generate-Cadl.ps1:29]:
```suggestion
Write-Host "Checking TypeSpec Java format ('npm run check-format')"
```

File: Generate-Cadl.ps1 [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+invokeExpressionAndCaptureOutput("npm run check-format")
+
+Write-Host "Packing CADL Java ('npm pack')"
```
Comment (haolingdong-msft) [inline Generate-Cadl.ps1:32]:
```suggestion
Write-Host "Packing TypeSpec Java ('npm pack')"
```

File: Generate-Cadl.ps1 [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+invokeExpressionAndCaptureOutput("npm pack")
+
+Write-Host "Returning to root directory ('..')"
+Set-Location ..
+
+Write-Host "Installing CADL ('npm install -g @cadl-lang/compiler')"
```
Comment (haolingdong-msft) [inline Generate-Cadl.ps1:38]:
```suggestion
Write-Host "Installing TypeSpec ('npm install -g @typespec/compiler')"
```

File: Generate-Cadl.ps1 [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+invokeExpressionAndCaptureOutput("npm install -g @cadl-lang/compiler")
```
Comment (haolingdong-msft) [inline Generate-Cadl.ps1:39]:
```suggestion
invokeExpressionAndCaptureOutput("npm install -g @typespec/compiler")
```

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
containerregistry changed again https://github.com/Azure/autorest.java/pull/2024

**Pull request merged**